velox/type/Timestamp.cpp

```diff
@@ -247,6 +247,7 @@ std::string Timestamp::tmToString(
   if (options.mode != TimestampToStringOptions::Mode::kTimeOnly) {
     int n = kTmYearBase + tmValue.tm_year;
     bool negative = n < 0;
+    const bool leadingPositiveSign = options.leadingPositiveSign && n > 9999;
```

Comment:
Is this a bug fix? Would it be possible to extract these changes into a separate PR with a unit test?

Reply:
It is to support Spark compatible behaviors through options. Opened https://github.com/facebookincubator/velox/pull/8165 for these change. And do you think we should extend the options or just add a new function for Spark in SparkCastHooks?

Reply:
@rui-mo Thank you for extracting this change into a separate PR. I left some comments there.

> And do you think we should extend the options or just add a new function for Spark in SparkCastHooks?

I don't have an opinion here. Extending options like you did seems fine to me.